extensions/wikia/InsightsV2/InsightsContext.class.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+class InsightsContext {
+	private $strategy = null;
+	private $config = null;
+	private $type = null;
+	private $articlesData = [];
+	private $params = [];
+	private $paginator = null;
+
+	public function __construct( InsightsModel $strategy, $type, $params ) {
+		$this->strategy = $strategy;
+		$this->config = $strategy->getConfig();
+		$this->type = $type;
+		$this->params = $params;
+	}
+
+	/**
+	 * Get list of articles related to the given QueryPage category
+	 *
+	 * @return array
+	 */
+	public function getContent() {
+		$content = [];
+
+		$this->articlesData = $this->fetchData();
+		if ( !empty ( $this->articlesData ) ) {
+			$sortedIds = $this->sortData();
+			$content = $this->sliceData( $sortedIds );
+		}
+
+		return $content;
+	}
+
+	public function fetchData() {
+		$cacheKey = ( new InsightsCache( $this->config ) )->getMemcKey( InsightsCache::INSIGHTS_MEMC_ARTICLES_KEY );
+		$articlesData = WikiaDataAccess::cache( $cacheKey, InsightsCache::INSIGHTS_MEMC_TTL, function () {
```

Comment:
InsightsCountService fixed here: https://github.com/Wikia/app/commit/4ecdde3fe59fec0a2d5573483051de943c9a2b76

Did you find any other places?